### PR TITLE
Change procedural loops to functions map() and filter()

### PR DESCRIPTION
**Pull request by gaugup**
## Description

This is follow up to the comment in PR https://github.com/microsoft/responsible-ai-toolbox/pull/1242 to change the procedural loops to functional style in typescript.

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [ ] responsibleai/core-ui
- [ ] responsibleai/counterfactuals
- [ ] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [ ] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [x] responsibleai/model-assessment

Python packages changed:

- [ ] raiwidgets
- [ ] responsibleai
- [ ] erroranalysis
- [ ] rai_core_flask

## Tests

- [x] No new tests required.
- [ ] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Documentation:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts [hunk 1]
```diff
@@ -118,14 +118,9 @@ function translatePreBuiltCohortFilterForTarget(
...
-    const index: number[] = [];
-    for (const modelClass of preBuiltCohortFilter.arg) {
-      const indexModelClass = modelClasses.indexOf(modelClass);
-
-      if (indexModelClass !== -1) {
-        index.push(indexModelClass);
-      }
-    }
+    const index = preBuiltCohortFilter.arg
+      .map((modelClass) => modelClasses.indexOf(modelClass))
```
Comment (imatiach-msft) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts:122]: might using the reduce function be even better?  
https://stackoverflow.com/questions/16037049/why-does-javascript-map-function-return-undefined

like this example they had:
```
var arr = ['a','b',1];
var results = arr.reduce((results, item) => {
    if (typeof item === 'string') results.push(modify(item)) // modify is a fictitious function that would apply some change to the items in the array
    return results
}, [])
```
map and filter also looks fine to me though

**Review by imatiach-msft — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts [hunk 2]
```diff
@@ -205,17 +197,13 @@ function translatePreBuiltCohortFilterForDataset(
...
       jointDataset.metaDict[jointDatasetFeatureName].sortedCategoricalValues;
     if (categorcialValues !== undefined) {
-      for (const categoricalValue of preBuiltCohortFilter.arg) {
-        const indexCategoricalValue =
-          categorcialValues.indexOf(categoricalValue);
-        if (indexCategoricalValue !== -1) {
-          index.push(indexCategoricalValue);
-        }
-      }
+      index = preBuiltCohortFilter.arg
```
Comment (imatiach-msft) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts:204]: I feel like all of these can just be refactored into one helper function.. although on the other hand I guess 3 lines of code isn't so much that it's really worth it to refactor perhaps?

Comment (gaugup) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts:204]: Yeah, map and filter are self explanatory constructs so may not need to define a function here.

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts [hunk 3]
```diff
@@ -205,17 +197,13 @@ function translatePreBuiltCohortFilterForDataset(
...
+        .map((categoricalValue) => categorcialValues.indexOf(categoricalValue))
```
Comment (romanlutz) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts:205]: nit: categorical instead of categorcial

**Review by romanlutz — APPROVED** [review]

**Pull request merged**